librosa/segment.py

```diff
@@ -797,7 +877,9 @@ def __my_filter(wrapped_f, *args, **kwargs):
 
 
 @cache(level=30)
-def subsegment(data, frames, *, n_segments=4, axis=-1):
+def subsegment(
+    data: np.ndarray, frames: np.ndarray, *, n_segments: int = 4, axis: int = -1
```

Comment:
Nope, ndarray is generic only from numpy 1.21 onwards. Even if our minimum version were 1.21, I wouldn't do any ndarray dtype restrictions in this pull request because that is a lot of work and is probably better done in another PR